main.tf

```diff
@@ -592,3 +595,54 @@ resource "aws_config_config_rule" "iam_mfa_for_console_access" {
     "aws_lambda_permission.iam_mfa_for_console_access",
   ]
 }
+
+### RESTRICTED COMMON PORTS: ACCESS
+resource "aws_config_config_rule" "restricted_common_ports_access" {
+  count = "${var.create_config_rules && ! local.exclude_restricted_common_ports_access ? 1 : 0}"
+
+  name             = "restricted-common-ports-access"
+  description      = "Checks whether security groups that are in use disallow unrestricted incoming TCP traffic to the specified ports."
+  input_parameters = "{\"blockedPort1\":\"22\",\"blockedPort2\":\"3389\"}"
+
+  scope {
+    compliance_resource_types = ["AWS::EC2::SecurityGroup"]
+  }
+
+  source {
+    owner             = "AWS"
+    source_identifier = "RESTRICTED_INCOMING_TRAFFIC"
+  }
+}
+
+### RESTRICTED COMMON PORTS: DATABASE
+resource "aws_config_config_rule" "restricted_common_ports_database" {
+  count = "${var.create_config_rules && ! local.exclude_restricted_common_ports_database ? 1 : 0}"
+
+  name             = "restricted-common-ports-database"
+  description      = "Checks whether security groups that are in use disallow unrestricted incoming TCP traffic to the specified ports."
+  input_parameters = "{\"blockedPort1\":\"1433\",\"blockedPort2\":\"1521\",\"blockedPort2\":\"3306\",\"blockedPort2\":\"4333\",\"blockedPort2\":\"5432\"}"
```

Comment:
Do the "blockedPort" numbers need to be different? 4 of these are set to `blockedPort2`

Reply:
Yeah, to terraform this is just a string, not really anything it can do to detect the error. The AWS API would need to throw the error, because that's where the string is json-decoded and where the duplicate keys could be detected. What usually happens in a dictionary with duplicate keys is that the last key wins.

Reply:
you're correct, last one won. latest config has all five values show up in the console.